### PR TITLE
Add support for linting GraphQL with ESLint

### PR DESCRIPTION
**Pull request by benjie**
Using https://github.com/apollographql/eslint-plugin-graphql, ESLint can be used to lint GraphQL files - this is good because it means the same lint rules can apply to GraphQL queries inline within your code as well as in dedicated files.

This PR basically just copies the `javascript/eslint.vim` linter and changes `javascript` for `graphql`.

### CODE REVIEW

File: doc/ale-graphql.txt [hunk 1]
```diff
@@ -2,6 +2,9 @@
 ALE GraphQL Integration                                   *ale-graphql-options*
 
 
+===============================================================================
+eslint                                                     *ale-graphql-eslint*
```
Comment (w0rp) [inline doc/ale-graphql.txt]: You should document that the JavaScript options are also used for GraphQL here. See `:help ale-typescript-eslint`.

Comment (benjie) [reply doc/ale-graphql.txt]: Sorry, I'm not familiar at all with writing vim docs - could you point me to an analogous location I can copy?

Comment (w0rp) [reply doc/ale-graphql.txt]: Yes, that help tag points to it. See `ale-typescript.txt`.

Comment (w0rp) [reply doc/ale-graphql.txt]: `*foo*` creates a help tag for `foo`, and `|foo|` references a help tag.

**Comment by benjie** [comment]
I wasn't sure quite how to word it so I just went with something simple. I also fixed a typo in the typescript docs; you're welcome to drop that commit if you want to keep things separate 👍

**Comment by benjie** [comment]
(Thanks so much for this plugin by the way, it's super useful 👍 )

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**